src/main/java/co/cask/http/HttpMethodInfo.java

```diff
@@ -110,6 +111,19 @@ void chunk(HttpChunk chunk) throws Exception {
     }
   }
 
+  void disconnected() {
+    if (bodyConsumer != null) {
+      bodyConsumerError(new ClosedChannelException());
+    }
+  }
+
+
+  void exceptionCaught(Throwable cause) {
```

Comment:
How will this method be invoked? Shouldn't ``HttpDispatcher`` override ``exceptionCaught``?